nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexFacetImpl.java

```diff
@@ -88,6 +94,43 @@ public void on(RepositoryDeletedEvent deletedEvent) {
     deleteIndexYaml();
   }
 
+  @Subscribe
+  @Guarded(by = STARTED)
+  @AllowConcurrentEvents
+  public void on(AssetCreateEvent created) {
+    log.debug("Updating index.yaml for hosted repository {}", getRepository().getName());
+    maybeInvalidateIndex(created);
+  }
+
+  @Subscribe
+  @Guarded(by = STARTED)
+  @AllowConcurrentEvents
+  public void on(AssetUpdateEvent updated) {
+    log.debug("Updating index.yaml for hosted repository {}", getRepository().getName());
+    maybeInvalidateIndex(updated);
+  }
+
+  @Subscribe
+  @Guarded(by = STARTED)
+  @AllowConcurrentEvents
+  public void on(AssetUploadEvent uploaded) {
+    log.debug("Updating index.yaml for hosted repository {}", getRepository().getName());
+    maybeInvalidateIndex(uploaded);
+  }
+
+  @Subscribe
+  @Guarded(by = STARTED)
+  @AllowConcurrentEvents
+  public void on(AssetPurgeEvent purged) {
+    log.debug("Updating index.yaml for hosted repository {}", getRepository().getName());
+    invalidateIndex();
+  }
+
+  private void maybeInvalidateIndex(final AssetEvent event) {
```

Comment:
Not sure but seems it is needed here one more check. Like:
if (getRepository().getName() == event.getAsset().getRepository())
could you confirm that this event processing will not work for all Helm repositories?